web/src/input_pill.ts

```diff
@@ -170,6 +170,11 @@ export function create<ItemType extends {type: string}>(
                 store.$parent.removeClass("invalid");
             }
 
+            // If we check is_pending just after adding a pill, the
+            // text is still present until further input, so we
+            // manually clear it here.
+            this.clear_text();
```

Comment:
Does this mean that if you past something invalid, like an email address, the thing you pasted disappears into the void?

Reply:
Seems to not do something bad in testing, so I'm a bit confused what this does.

Reply:
Oh, this is the typeahead code path, fine.